Signal/src/views/ContactTableViewCell.m

```diff
@@ -60,19 +61,37 @@ - (void)configureProgrammatically
     _avatarView = [AvatarImageView new];
     [self.contentView addSubview:_avatarView];
 
+    UIView *nameContainerView = [UIView containerView];
+    [self.contentView addSubview:nameContainerView];
+
     _nameLabel = [UILabel new];
     _nameLabel.lineBreakMode = NSLineBreakByTruncatingTail;
     _nameLabel.font = [UIFont ows_dynamicTypeBodyFont];
-    [self.contentView addSubview:_nameLabel];
+    [nameContainerView addSubview:_nameLabel];
+
+    _profileNameLabel = [UILabel new];
+    _profileNameLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+    _profileNameLabel.font = [UIFont ows_footnoteFont];
+    _profileNameLabel.textColor = [UIColor grayColor];
+    [nameContainerView addSubview:_profileNameLabel];
 
     [_avatarView autoVCenterInSuperview];
     [_avatarView autoPinLeadingToSuperView];
     [_avatarView autoSetDimension:ALDimensionWidth toSize:kContactTableViewCellAvatarSize];
     [_avatarView autoSetDimension:ALDimensionHeight toSize:kContactTableViewCellAvatarSize];
 
-    [_nameLabel autoVCenterInSuperview];
-    [_nameLabel autoPinLeadingToTrailingOfView:_avatarView margin:12.f];
-    [_nameLabel autoPinTrailingToSuperView];
+    [_nameLabel autoPinEdgeToSuperviewEdge:ALEdgeTop];
+    [_nameLabel autoPinWidthToSuperview];
+
+    // profileNameLabel can be zero sized, in which case nameLabel essentially occupies the totality of
+    // nameContainerView's frame.
+    [_profileNameLabel autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:_nameLabel];
+    [_profileNameLabel autoPinEdgeToSuperviewEdge:ALEdgeBottom];
+    [_profileNameLabel autoPinWidthToSuperview];
+
+    [nameContainerView autoVCenterInSuperview];
+    [nameContainerView autoPinLeadingToTrailingOfView:_avatarView margin:12.f];
+    [nameContainerView autoPinTrailingToSuperView];
```

Comment:
Doesn't `ContactTableViewCell` sometimes use accessory views, e.g. "not verified" in your screenshots?  Shouldn't we use a "manual" accessory view and pin trailing to that, with margin?